### ISSUE TITLE
A1H Solution Testing - Prod/Test External

### ISSUE THREAD

**Issue by rhood23699** [CDCgov/prime-reportstream#3245]
**Description:**
Complete full Integrated Test with the State Public Health Department.

**Main Objectives & Tasks:**

- [x] Contact the State Public Health Department to setup testing with Production data in the Production environment.
- [x] Work directly with the State Public Health Department to complete Integrated Testing and gain approval to move into Production.

**Comment by rhood23699** [comment]
Current plans are to address steps required to begin A1H testing on the Thursday, 12/2/21 weekly sync call with CA.

**Comment by rhood23699** [comment]
12/2/21 - Had our kickoff meeting with CA for A1H testing.  Summary is:
- CA will send results to LA and San Diego counties.
- CA will require us to use the new centralized CLIA of CDPH000087.  I'll coordinate with @brick-green-agile6 on this.
- CA wants to results going to their test environment.  @brick-green-agile6 will help me with this.
- I must change the schema to use a default provider if the Provider ID/ NPI is blank.
- Our target for the 1st production file for testing to CA is Tuesday, Dec 7th.  This is dependent on my schema changes making it into PROD and @brick-green-agile6 helping me with the centralized CLIA and routing of results to their test environment.

**Comment by rhood23699** [comment]
Committed to providing a list of required fields based on the existing A1H CSV file to help Simmi make sure she enters all required data on teh 20% of results that she enters by hand.

**Comment by rhood23699** [comment]
- Postponed Tuesday, Dec 7th first PROD file upload to CA's test environment.  There was no Deployment last Thursday or Friday.

- Provided Simmi with a version of her CSV file indicating which fields are required.  She needed this because 20% of her results will require manual entry, so this will help her make sure the data is correct.

- Started schema changes for adding the default provider.

**Comment by rhood23699** [comment]
Email'd CA Dept Health about which CLIA# they want us to use for CSV file based results.  Also, working with Brick to setup new Feed to sent Prod/Test results to CA test environment.

**Comment by rhood23699** [comment]
Today's deployment is delayed until tomorrow the 8th.  The changes to add the default provider is scheduled for Thursday.

Ticket#: 3338 - @brick-green-agile6 is working on this ticket to add a new CA-test feed to capture and send all processing mode "T" and "D" to CA's test environment.  This should be completed by the Thursday 12/8/21 deployment.

**Comment by rhood23699** [comment]
Our system shows the file was send via SFTP successfully, but CA doesn't have it.  I've asked Brick to verify that it was sent, and to verify if central CLIA#: CDPH000087 was used.

**Comment by rhood23699** [comment]
12/10/21 - Brick verified my findings that the files are being sent.  CA responded late Friday that there is a delay on their side and that Optum may be able to verify if the results were received, and if there are any errors.

**Comment by rhood23699** [comment]
Optum verified that they are receiving the test file, but it is failing because the MSH.3.1 needs to be truncated to 20 characters.  @brick-green-agile6 is researching.

**Comment by rhood23699** [comment]
Brick has made the change to truncate the MSH.3.1 to 20 characters.  The test file was resent about 12p.  Email'd Optum twice and waiting on verification that they received the data.

**Comment by rhood23699** [comment]
3:36p CT - Vanessa confirmed receipt of the results.  She found missing patient phone #s.  We didn't send phone #s, so this is not a problem.  She received received all 5 results in Rhapsody, but only 4 made it into CalREDIE, so she is researching why.

**Comment by rhood23699** [comment]
Email'd CA regarding status on identifying the problem with the 1 result.  They are working on it, and stated they would reply in our email thread when they finished.

**Comment by mlighdc** [comment]
@rhood23699 thanks for all the updates in github. We have a call with CA tomorrow. I am sure the CSV upload topic will come up if we have not heard from them. Can you please make sure I am cc-ed on the emails as I don't see the email thread?

**Comment by rhood23699** [comment]
12/15/21 - CA confirmed that all results were accepted.  I email'd A1J asking them to send a Production file with PHI.

**Comment by rhood23699** [comment]
The test file failed with blank Patient County.  I'll modify the schema to allow blank Patient Counties.  I'll check with @brick-green-agile6 on whether or not the qualityFilter fail with a blank County.  CA says they will accept blank County.  I asked A1H to go ahead and send another test file even before the County change is made.

**Comment by rhood23699** [comment]
Waiting on the Patient County changes to be deployed, so I can ask Simmi to resend file.  Simmi as of right now, hasn't sent another test file.

**Comment by rhood23699** [comment]
Email'd Simmi asking if she will be able to send a file today?

**Comment by rhood23699** [comment]
I've asked Ming to contact Simmi about sending another test file.  Ming will let her know that we need a file after 1p ET tomorrow.  Ming believes Simmi may be able to send a file tonight, but Simmi may be traveling.

**Comment by rhood23699** [comment]
A1H sent a file last night, 12/20/21.  11 results were sent, but only 3 made it to CA.  I've email'd asking for the de-identified version of the A1H test file to confirm that an invalid equipment model name was the issue.  I've update @carlosfelix2 and @MauriceReeves-usds that this A1H test appears to confirm early findings outlined in Ticket #2616 stating that an invalid equipment model name doesn't show as an error to the Sender.  I'll followup to make sure everyone understands that this is a roadblock to moving A1H to production.

**Comment by rhood23699** [comment]
12/21/22 @jimduff-usds contacted me about the equipment model name problem.  We'll start work on this tomorrow at 12/23/21.

**Comment by rhood23699** [comment]
Simmi sent the de-identified file.  It confirms that the equipment model name was invalid on the 3 results.  She also stated that she would like our system to note only inform her of the problem, but block the entire file from processing not just the records that failed.  She states it will be easier for her to fix the records in the existing file and resubmit the entire file rather than create a new file with only the revised rows that error'd out.

**Comment by rhood23699** [comment]
A1H Update - Met with Carlos and JimD about the equipment model name problem.  Good and bad news.  First the good news, JimD gave me a schema workaround to make sure A1H files fail if the Equipment Model Name is blank.  I'll make the change and create a PR today.  Bad news, there is a bug and no quick fix for it.  If the Equipment Mode Name is invalid, A1H doesn't receive an error message; therefore neither A1H or CA knows results are missing.  There is a temporary workaround that would require someone to monitor A1H submission in Metabase; review Actions and identify the results excluded from the CA submission; and then contact A1H to have them re-submit only the rows with errors.
A1H test files have shown that most of the time when there is a bad equipment model name, it is caused because the value is blank; therefore A1H will receive the error message and be able to correct and re-send the file.  The problem is A1H does perform manual entry for some of their results, so if we proceed with testing and moving A1H to Production, someone will have to monitor Metabase and report errors to A1H, and A1H will have to agree to this temporary process.

**Comment by rhood23699** [comment]
12/23/21 - Both CAPHD and A1H agreed to move into production with the temporary solution of rejected blank equipment model name and then manually monitoring for invalid equipment model.  Closing this ticket and start Planning GO LIVE>
